x/data/iri.go

```diff
@@ -0,0 +1,113 @@
+package data
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/btcsuite/btcutil/base58"
+)
+
+// AccAddressToDID converts an account address to a DID using a chain-specific method prefix,
+// which should generally be equivalent to the chain's bech32 account prefix.
+func AccAddressToDID(address sdk.AccAddress, methodPrefix string) string {
+	hash := base58.CheckEncode(address, didVersion0)
+	return fmt.Sprintf("did:%s:%s", methodPrefix, hash)
+}
+
+// ToIRI converts the ContentHash to an IRI (internationalized URI) using the regen IRI scheme.
+// A ContentHash IRI will look something like regen:113gdjFKcVCt13Za6vN7TtbgMM6LMSjRnu89BMCxeuHdkJ1hWUmy.rdf
+// which is some base58check encoded data followed by a file extension or pseudo-extension.
+// See ContentHash_Raw.ToIRI and ContentHash_Graph.ToIRI for more details on specific formatting.
+func (ch ContentHash) ToIRI() (string, error) {
+	switch hash := ch.Sum.(type) {
+	case *ContentHash_Raw_:
+		return hash.Raw.ToIRI()
+	case *ContentHash_Graph_:
+		return hash.Graph.ToIRI()
+	default:
+		return "", fmt.Errorf("invalid %T type %T", ch, hash)
+	}
+}
```

Comment:
Why do we call it `IRI` instead of `URI`? We use only ASCII characters, so the output is always `URI`.

Reply:
Because IRI is the terminology used in all RDF contexts